Formula/onefetch.rb

```diff
@@ -0,0 +1,17 @@
+class Onefetch < Formula
+  desc "Git repository summary on your terminal"
+  homepage "https://github.com/o2sh/onefetch"
+  url "https://github.com/o2sh/onefetch/archive/v1.6.5.tar.gz"
+  sha256 "50f069db2fa713024a19bf708add7812f3420b0c107eb68cd9907d76e375a06d"
+
+  depends_on "rust" => :build
+
+  def install
+    system "cargo", "install", "--root", prefix, "--path", "."
+  end
+
+  test do
+    system "#{bin}/onefetch", "--help"
+    assert_match "onefetch " + version.to_s, shell_output("#{bin}/onefetch -V").chomp
```

Comment:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.

Reply:
Hi @SMillerDev, thanks for your review.

The only functionality we can test here is the execution of `onefetch .`, which throws the following error (only on `stderr`): `Error: Could not find any source code in this directory`. Unfortunately, `shell_output` only returns the output to `stdout`, but allows the assertion of the exit code.

As we don't have a git repo in the testing sandbox, is adding the following assertion sufficient?

```ruby
shell_output("#{bin}/onefetch", 1)
```

Otherwise, we could setup an empty git repository somewhere, and test the output of the command there. But I'm not sure if that would be according to the Formula guidelines.

```ruby
shell_output("git init && echo \"puts 'Hello, world'\" > main.rb && git add main.rb && git commit -m \"First commit\"")
assert_match "Ruby", shell_output("#{bin}/onefetch").chomp
```

Reply:
The better your test is, the closer it follows the guidelines 😁. I'd probably just use `system "some command"` to set up the repo though.

Reply:
You could even check out Homebrew's onefetch output:

```
system "#{bin}/onefetch", "-d", "#{HOMEBREW_PREFIX}/Homebrew"
```

Of course you'll want to check the output makes sense... This may also add complications with Linux brew if you're not careful about what you look for in the output.

Reply:
I wouldn't risk doing stuff outside the container/temporary directory.